ludwig/utils/commit_utils.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import os
+from typing import Optional
+
+from huggingface_hub import HfApi, login
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceHub:
+    def __init__(self):
+        self.api = None
+
+    def login(self):
+        """Login to huggingface hub using the token stored in ~/.cache/huggingface/token and returns a HfApi client
+        object that can be used to interact with HF Hub."""
+        cached_token_path = os.path.join(os.path.expanduser("~"), ".cache", "huggingface", "token")
+
+        if not os.path.exists(cached_token_path):
+            login(add_to_git_credential=True)
+
+        with open(cached_token_path) as f:
+            hf_token = f.read()
+
+        hf_api = HfApi(token=hf_token)
+        assert hf_api.token == hf_token
+
+        self.api = hf_api
+
+    def upload_to_hfhub(
```

Comment:
one idea here might be to have a flag that additionally allows to publish to our model-hub or maybe we should discuss how the two should work together

Reply:
Agreed. We can define a base class and expand the arguments to be:

1. `ludwig commit hf_hub ...`
2. `ludwig commit ludwig_hub ...`